### ISSUE TITLE
[core] Actor processes spam stdout on kill when #actors > #cpus

### ISSUE THREAD

**Issue by johnoleary** [ray-project/ray#16613]
### What is the problem?

Using the latest Ray nightly on Linux Python 3.7:

When the number of actor processes exceeds the number of CPUs given by ray.init(num_cpus=...), later actor processes print an empty (pid=...) line to stdout on termination. In particular if there are X CPUs, actor processes 1, ..., X terminate normally, i.e. without printing anything, while processes X + 1 and beyond print an empty line on termination. This behavior does not depend on whether the actor processes come from the same actor class or different ones.

### Reproduction (REQUIRED)

```
import ray
import time

ray.init(num_cpus=1)

@ray.remote
class Actor:
    pass

a = Actor.remote()  # actor process 1 <= num_cpus
b = Actor.remote()  # actor process 2 > num_cpus

ray.kill(a)  # silent termination as expected
ray.kill(b)  # prints "(pid=...)" to stdout

time.sleep(5)  # print takes a second or two to appear
ray.shutdown()
```

- [X] I have verified my script runs in a clean environment and reproduces the issue.
- [X] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

**Comment by ericl** [comment]
I cannot reproduce this (Linux, py37). @richardliaw @wuisawesome can you?

**Comment by richardliaw** [comment]
Cannot repro on 3.7 mac. Trying again on latest master

**Comment by richardliaw** [comment]
Cannot repro on latest master.. for 3.7 mac unfortunately

**Comment by richardliaw** [comment]
@johnoleary sorry to bother; can you reproduce this on again with the latest nightly wheel?

i.e. -- `ray uninstall ray; pip install -U --no-cache-dir https://s3-us-west-2.amazonaws.com/ray-wheels/latestray-2.0.0.dev0-cp37-cp37m-manylinux2014_x86_64.whl`?

**Comment by johnoleary** [comment]
No problem at all.  On RHEL 7.9, I created a fresh environment and installed the latest wheel via `pip install -U --no-cache-dir https://s3-us-west-2.amazonaws.com/ray-wheels/latest/ray-2.0.0.dev0-cp37-cp37m-manylinux2014_x86_64.whl`. I'm glad to report that everything looks good.

Back in the environment for my current project, I installed the latest wheel as above. Here the stdout issue remains, so it seems like it might be due to an interaction between ray and something else in the environment. I'll dig into this and see if I can come up with something generally reproducible.
